### PR TITLE
Adds getting started docs for macOS

### PR DESCRIPTION
**Pull request by anthgur**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/413)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,55 @@
...
+
+```sh
+rustup toolchain install nightly
+rustup default nightly
+```
+
+### Other dependencies
+
+#### OS X
+#### On OS X (homebrew)
```
Comment (atouchet) [inline README.md]: Apple no longer uses the OS X branding. Can these be changed to say macOS instead? Also I think homebrew should be capitalized here.

Comment (anthgur) [reply README.md]: Good catch, thank you!

File: README.md [hunk 2]
```diff
@@ -3,3 +3,55 @@
 Rust bindings to SpiderMonkey
 
 [Documentation](http://doc.servo.org/mozjs/)
+
+## Setting up your environment
+
+### Rust
+
+This project requires Rust nightly-2018-03-25 or greater. You can install this with Rustup.rs:
```
Comment (jdm) [inline README.md]: I don't think it makes sense to repeat the rustup installation instructions here. It's more efficient to direct people to https://rustup.rs/.

Comment (anthgur) [reply README.md]: That makes sense, can we keep the nightly info though? I spent some time figuring out I needed to use nightly instead of stable.

Comment (jdm) [reply README.md]: Yes, that's fine.

**Comment by jdm** [comment]
@bors-servo retry
* #416

**Pull request merged**